omnigibson/action_primitives/starter_semantic_action_primitives.py

```diff
@@ -1414,24 +1418,17 @@ def _empty_action(self):
         """
         action = np.zeros(self.robot.action_dim)
         for name, controller in self.robot._controllers.items():
-            joint_idx = controller.dof_idx
             action_idx = self.robot.controller_action_idx[name]
-            if (
-                controller.control_type == ControlType.POSITION
-                and len(joint_idx) == len(action_idx)
-                and not controller.use_delta_commands
-            ):
-                action[action_idx] = self.robot.get_joint_positions()[joint_idx]
-            elif self.robot._controller_config[name]["name"] == "InverseKinematicsController":
-                # overwrite the goal orientation, since it is in absolute frame.
-                assert (
-                    self.robot._controller_config["arm_" + self.arm]["mode"] == "pose_absolute_ori"
-                ), "Controller must be in pose_absolute_ori mode"
-                current_quat = self.robot.get_relative_eef_orientation()
-                current_ori = T.quat2axisangle(current_quat)
-                control_idx = self.robot.controller_action_idx["arm_" + self.arm]
-                action[control_idx[3:]] = current_ori
+            no_op_goal = controller.compute_no_op_goal(self.robot.get_control_dict())
+
+            if self.robot._controller_config[name]["name"] == "InverseKinematicsController":
+                # assert (
+                #     self.robot._controller_config["arm_" + self.arm]["mode"] == "pose_absolute_ori"
+                # ), "Controller must be in pose_absolute_ori mode"
+                # convert quaternion to axis-angle representation for control input
+                no_op_goal["target_quat"] = T.quat2axisangle(no_op_goal["target_quat"])
```

Comment:
why is it needed to convert quaternion to axis-angle representation for control input?

Reply:
I suppose this is a thing with IK controller `pose_absolute_ori` mode. It seems to me that it takes axis angles as action. This might help:
```
"pose_absolute_ori": 6,  # 6DOF (dx,dy,dz,ax,ay,az) control over pose, where the orientation is given in absolute axis-angle coordinates
```

Reply:
move the logic into `InverseKinematicsController`

Reply:
both OSC and IK take in axis angle, so their action spaces were unified in this way.

It's a bit more efficient than using quaternions, since it's 3-dim vs. 4-dim

Reply:
@cremebrule why axis angle instead of euler angles? iGibson uses to do euler I think and so does the joint controlled base on the tiago etc. Maybe not the best place to discuss this but this is the first time I'm hearing we use axis angle representation anywhere so just curious